Multiple test_project.py::test_search tests fail due to wrong order

Not sure if jedi is supposed to return results in a specific order, or if sorting needs to be added to the test to account for different return order:

```
=============================================================== test session starts ===============================================================
platform linux -- Python 3.8.2, pytest-5.4.1, py-1.8.0, pluggy-0.13.1 -- /usr/bin/python3.8
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/tmp/jedi/.hypothesis/examples')
rootdir: /tmp/jedi, inifile: pytest.ini
plugins: expect-1.1.0, django-3.8.0, requests-mock-1.7.0, aiohttp-0.3.0, betamax-0.8.1, httpbin-1.0.0, mock-1.11.2, shutil-1.7.0, virtualenv-1.7.0, timeout-1.3.3, xprocess-0.13.1, flaky-3.6.1, hypothesis-5.8.3
collected 1 item                                                                                                                                  

test/test_api/test_project.py::test_search[implicit_namespace_package.-full_names22-kwargs22] FAILED                                        [100%]

==================================================================== FAILURES =====================================================================
_________________________________________ test_search[implicit_namespace_package.-full_names22-kwargs22] __________________________________________

string = 'implicit_namespace_package.', full_names = ['examples.implicit_namespace_package.ns1', 'examples.implicit_namespace_package.ns2']
kwargs = {}, skip_pre_python36 = None

    @pytest.mark.parametrize(
        'string, full_names, kwargs', [
            ('test_load_save_project', ['test_api.test_project.test_load_save_project'], {}),
            ('test_load_savep', [], dict(complete=True)),
            ('test_load_save_p', ['test_api.test_project.test_load_save_project'],
             dict(complete=True)),
            ('test_load_save_p', ['test_api.test_project.test_load_save_project'],
             dict(complete=True, all_scopes=True)),
    
            ('some_search_test_var', [], {}),
            ('some_search_test_var', ['test_api.test_project.test_search.some_search_test_var'],
             dict(all_scopes=True)),
            ('some_search_test_var', ['test_api.test_project.test_search.some_search_test_var'],
             dict(complete=True, all_scopes=True)),
    
            ('sample_int', ['helpers.sample_int'], {}),
            ('sample_int', ['helpers.sample_int'], dict(all_scopes=True)),
            ('sample_int.real', ['stub:builtins.int.real'], {}),
    
            ('class sample_int.real', [], {}),
            ('foo sample_int.real', [], {}),
            ('def sample_int.real', ['stub:builtins.int.real'], {}),
            ('function sample_int.real', ['stub:builtins.int.real'], {}),
    
            # With modules
            ('test_project.test_search', ['test_api.test_project.test_search'], {}),
            ('test_project.test_searc', ['test_api.test_project.test_search'], dict(complete=True)),
            ('test_api.test_project.test_search', ['test_api.test_project.test_search'], {}),
            ('test_api.test_project.test_sear', ['test_api.test_project.test_search'],
             dict(complete=True)),
    
            # With namespace
            ('implicit_namespace_package.ns1.pkg',
             ['examples.implicit_namespace_package.ns1.pkg'], {}),
            ('implicit_namespace_package.ns1.pkg.ns1_file',
             ['examples.implicit_namespace_package.ns1.pkg.ns1_file'], {}),
            ('examples.implicit_namespace_package.ns1.pkg.ns1_file',
             ['examples.implicit_namespace_package.ns1.pkg.ns1_file'], {}),
            ('implicit_namespace_package.ns1.pkg.',
             ['examples.implicit_namespace_package.ns1.pkg.ns1_file'],
             dict(complete=True)),
            ('implicit_namespace_package.',
             ['examples.implicit_namespace_package.ns1',
              'examples.implicit_namespace_package.ns2'],
             dict(complete=True)),
    
            # With stubs
            ('with_python.module', ['examples.stub_packages.with_python.module'], {}),
            ('with_python.modul', ['examples.stub_packages.with_python.module'],
             dict(complete=True)),
            ('no_python.foo', ['stub:examples.stub_packages.no_python.foo'], {}),
            ('no_python.fo', ['stub:examples.stub_packages.no_python.foo'],
             dict(complete=True)),
            ('with_python-stubs.module', [], {}),
            ('no_python-stubs.foo', [], {}),
            # Both locations are given, because they live in separate folders (one
            # suffixed with -stubs.
            ('with_python', ['examples.stub_packages.with_python'], {}),
            ('no_python', ['stub:examples.stub_packages.no_python'], {}),
            # Completion stubs
            ('stub_only', ['stub:completion.stub_folder.stub_only',
                           'stub:examples.stub_packages.with_python.stub_only'], {}),
            ('with_stub', ['completion.stub_folder.with_stub'], {}),
            ('with_stub.in_with_stub_both',
             ['completion.stub_folder.with_stub.in_with_stub_both'], {}),
            ('with_stub.in_with_stub_python',
             ['completion.stub_folder.with_stub.in_with_stub_python'], {}),
            ('with_stub.in_with_stub_stub',
             ['stub:completion.stub_folder.with_stub.in_with_stub_stub'], {}),
            # Completion stubs: Folder
            ('with_stub_folder', ['completion.stub_folder.with_stub_folder'], {}),
            ('with_stub_folder.nested_with_stub',
             ['completion.stub_folder.with_stub_folder.nested_with_stub'], {}),
            ('nested_with_stub',
             ['completion.stub_folder.stub_only_folder.nested_with_stub',
              'completion.stub_folder.with_stub_folder.nested_with_stub'], {}),
    
            # On sys path
            ('sys.path', ['stub:sys.path'], {}),
            ('json.dumps', ['json.dumps'], {}),  # stdlib + stub
            ('multiprocessing', ['multiprocessing'], {}),
            ('multiprocessin', ['multiprocessing'], dict(complete=True)),
        ]
    )
    @pytest.mark.skipif(sys.version_info < (3, 6), reason="Ignore Python 2, because EOL")
    def test_search(string, full_names, kwargs, skip_pre_python36):
        some_search_test_var = 1.0
        project = Project(test_dir)
        if kwargs.pop('complete', False) is True:
            defs = project.complete_search(string, **kwargs)
        else:
            defs = project.search(string, **kwargs)
>       assert [('stub:' if d.is_stub() else '') + d.full_name for d in defs] == full_names
E       AssertionError: assert ['examples.implicit_namespace_package.ns2',\n 'examples.implicit_namespace_package.ns1'] == ['examples.implicit_namespace_package.ns1',\n 'examples.implicit_namespace_package.ns2']
E         At index 0 diff: 'examples.implicit_namespace_package.ns2' != 'examples.implicit_namespace_package.ns1'
E         Full diff:
E           [
E         +  'examples.implicit_namespace_package.ns2',
E            'examples.implicit_namespace_package.ns1',
E         -  'examples.implicit_namespace_package.ns2',
E           ]

test/test_api/test_project.py:138: AssertionError
============================================================= short test summary info =============================================================
FAILED test/test_api/test_project.py::test_search[implicit_namespace_package.-full_names22-kwargs22] - AssertionError: assert ['examples.implici...
================================================================ 1 failed in 2.99s ================================================================

```

Also this one:

```
=============================================================== test session starts ===============================================================
platform linux -- Python 3.8.2, pytest-5.4.1, py-1.8.0, pluggy-0.13.1 -- /usr/bin/python3.8
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/tmp/jedi/.hypothesis/examples')
rootdir: /tmp/jedi, inifile: pytest.ini
plugins: expect-1.1.0, django-3.8.0, requests-mock-1.7.0, aiohttp-0.3.0, betamax-0.8.1, httpbin-1.0.0, mock-1.11.2, shutil-1.7.0, virtualenv-1.7.0, timeout-1.3.3, xprocess-0.13.1, flaky-3.6.1, hypothesis-5.8.3
collected 1 item                                                                                                                                  

test/test_api/test_project.py::test_search[nested_with_stub-full_names38-kwargs38] FAILED                                                   [100%]

==================================================================== FAILURES =====================================================================
_______________________________________________ test_search[nested_with_stub-full_names38-kwargs38] _______________________________________________

string = 'nested_with_stub'
full_names = ['completion.stub_folder.stub_only_folder.nested_with_stub', 'completion.stub_folder.with_stub_folder.nested_with_stub'], kwargs = {}
skip_pre_python36 = None

    @pytest.mark.parametrize(
        'string, full_names, kwargs', [
            ('test_load_save_project', ['test_api.test_project.test_load_save_project'], {}),
            ('test_load_savep', [], dict(complete=True)),
            ('test_load_save_p', ['test_api.test_project.test_load_save_project'],
             dict(complete=True)),
            ('test_load_save_p', ['test_api.test_project.test_load_save_project'],
             dict(complete=True, all_scopes=True)),
    
            ('some_search_test_var', [], {}),
            ('some_search_test_var', ['test_api.test_project.test_search.some_search_test_var'],
             dict(all_scopes=True)),
            ('some_search_test_var', ['test_api.test_project.test_search.some_search_test_var'],
             dict(complete=True, all_scopes=True)),
    
            ('sample_int', ['helpers.sample_int'], {}),
            ('sample_int', ['helpers.sample_int'], dict(all_scopes=True)),
            ('sample_int.real', ['stub:builtins.int.real'], {}),
    
            ('class sample_int.real', [], {}),
            ('foo sample_int.real', [], {}),
            ('def sample_int.real', ['stub:builtins.int.real'], {}),
            ('function sample_int.real', ['stub:builtins.int.real'], {}),
    
            # With modules
            ('test_project.test_search', ['test_api.test_project.test_search'], {}),
            ('test_project.test_searc', ['test_api.test_project.test_search'], dict(complete=True)),
            ('test_api.test_project.test_search', ['test_api.test_project.test_search'], {}),
            ('test_api.test_project.test_sear', ['test_api.test_project.test_search'],
             dict(complete=True)),
    
            # With namespace
            ('implicit_namespace_package.ns1.pkg',
             ['examples.implicit_namespace_package.ns1.pkg'], {}),
            ('implicit_namespace_package.ns1.pkg.ns1_file',
             ['examples.implicit_namespace_package.ns1.pkg.ns1_file'], {}),
            ('examples.implicit_namespace_package.ns1.pkg.ns1_file',
             ['examples.implicit_namespace_package.ns1.pkg.ns1_file'], {}),
            ('implicit_namespace_package.ns1.pkg.',
             ['examples.implicit_namespace_package.ns1.pkg.ns1_file'],
             dict(complete=True)),
            ('implicit_namespace_package.',
             ['examples.implicit_namespace_package.ns1',
              'examples.implicit_namespace_package.ns2'],
             dict(complete=True)),
    
            # With stubs
            ('with_python.module', ['examples.stub_packages.with_python.module'], {}),
            ('with_python.modul', ['examples.stub_packages.with_python.module'],
             dict(complete=True)),
            ('no_python.foo', ['stub:examples.stub_packages.no_python.foo'], {}),
            ('no_python.fo', ['stub:examples.stub_packages.no_python.foo'],
             dict(complete=True)),
            ('with_python-stubs.module', [], {}),
            ('no_python-stubs.foo', [], {}),
            # Both locations are given, because they live in separate folders (one
            # suffixed with -stubs.
            ('with_python', ['examples.stub_packages.with_python'], {}),
            ('no_python', ['stub:examples.stub_packages.no_python'], {}),
            # Completion stubs
            ('stub_only', ['stub:completion.stub_folder.stub_only',
                           'stub:examples.stub_packages.with_python.stub_only'], {}),
            ('with_stub', ['completion.stub_folder.with_stub'], {}),
            ('with_stub.in_with_stub_both',
             ['completion.stub_folder.with_stub.in_with_stub_both'], {}),
            ('with_stub.in_with_stub_python',
             ['completion.stub_folder.with_stub.in_with_stub_python'], {}),
            ('with_stub.in_with_stub_stub',
             ['stub:completion.stub_folder.with_stub.in_with_stub_stub'], {}),
            # Completion stubs: Folder
            ('with_stub_folder', ['completion.stub_folder.with_stub_folder'], {}),
            ('with_stub_folder.nested_with_stub',
             ['completion.stub_folder.with_stub_folder.nested_with_stub'], {}),
            ('nested_with_stub',
             ['completion.stub_folder.stub_only_folder.nested_with_stub',
              'completion.stub_folder.with_stub_folder.nested_with_stub'], {}),
    
            # On sys path
            ('sys.path', ['stub:sys.path'], {}),
            ('json.dumps', ['json.dumps'], {}),  # stdlib + stub
            ('multiprocessing', ['multiprocessing'], {}),
            ('multiprocessin', ['multiprocessing'], dict(complete=True)),
        ]
    )
    @pytest.mark.skipif(sys.version_info < (3, 6), reason="Ignore Python 2, because EOL")
    def test_search(string, full_names, kwargs, skip_pre_python36):
        some_search_test_var = 1.0
        project = Project(test_dir)
        if kwargs.pop('complete', False) is True:
            defs = project.complete_search(string, **kwargs)
        else:
            defs = project.search(string, **kwargs)
>       assert [('stub:' if d.is_stub() else '') + d.full_name for d in defs] == full_names
E       AssertionError: assert ['completion.stub_folder.with_stub_folder.nested_with_stub',\n 'completion.stub_folder.stub_only_folder.nested_with_stub'] == ['completion.stub_folder.stub_only_folder.nested_with_stub',\n 'completion.stub_folder.with_stub_folder.nested_with_stub']
E         At index 0 diff: 'completion.stub_folder.with_stub_folder.nested_with_stub' != 'completion.stub_folder.stub_only_folder.nested_with_stub'
E         Full diff:
E           [
E         +  'completion.stub_folder.with_stub_folder.nested_with_stub',
E            'completion.stub_folder.stub_only_folder.nested_with_stub',
E         -  'completion.stub_folder.with_stub_folder.nested_with_stub',
E           ]

test/test_api/test_project.py:138: AssertionError
============================================================= short test summary info =============================================================
FAILED test/test_api/test_project.py::test_search[nested_with_stub-full_names38-kwargs38] - AssertionError: assert ['completion.stub_folder.with...
================================================================ 1 failed in 2.42s ================================================================

```

--- a/test/test_api/test_project.py
+++ b/test/test_api/test_project.py
@@ -135,7 +135,7 @@ def test_search(string, full_names, kwargs, skip_pre_python36):
         defs = project.complete_search(string, **kwargs)
     else:
         defs = project.search(string, **kwargs)
-    assert [('stub:' if d.is_stub() else '') + d.full_name for d in defs] == full_names
+    assert sorted([('stub:' if d.is_stub() else '') + d.full_name for d in defs]) == full_names
 
 
 @pytest.mark.parametrize(

